File: src/backend/schemas/food.py
"""식품 관련 API 스키마."""

from pydantic import BaseModel


class FoodSearchItem(BaseModel):
    """식품 검색 결과 아이템."""

    id: int
    name: str
    slug: str
    category: str | None = None
    image_url: str | None = None

    model_config = {"from_attributes": True}


class FoodDetail(BaseModel):
    """식품 상세 정보."""

    id: int
    name: str
    slug: str
    category: str | None = None
    description: str | None = None
    common_names: list[str] | None = None
    nutrients: list[dict] | None = None
    image_url: str | None = None

    model_config = {"from_attributes": True}


class FoodSearchResponse(BaseModel):
    """식품 검색 응답."""

    items: list[FoodSearchItem]
    total: int
    page: int
    page_size: int

File: src/web/src/types/food.ts
/** 식품 관련 타입 정의 — 백엔드 스키마 1:1 매칭. */

export interface FoodSearchItem {
  id: number;
  name: string;
  slug: string;
  category: string | null;
  image_url: string | null;
}

export interface FoodDetail {
  id: number;
  name: string;
  slug: string;
  category: string | null;
  description: string | null;
  common_names: string[] | null;
  nutrients: Record<string, unknown>[] | null;
  image_url: string | null;
}
